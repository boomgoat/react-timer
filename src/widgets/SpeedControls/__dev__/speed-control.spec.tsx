import React from "react";
import { render, screen } from "@testing-library/react";
import { SpeedControl } from "..";

describe("Speed Control Tests", () => {
  const setSpeed = () => {};

  test("speed set to 1.0x", () => {
    const speed: number = 1;
    render(<SpeedControl speed={speed} setSpeed={setSpeed} />);

    expect(screen.getByTestId("1.0x-button")).toHaveClass("bg-green-900");
    expect(screen.getByTestId("1.5x-button")).not.toHaveClass("bg-green-900");
    expect(screen.getByTestId("2.0x-button")).not.toHaveClass("bg-green-900");
  });

  test("speed set to 1.5x", () => {
    const speed: number = 1.5;
    render(<SpeedControl speed={speed} setSpeed={setSpeed} />);

    expect(screen.getByTestId("1.0x-button")).not.toHaveClass("bg-green-900");
    expect(screen.getByTestId("1.5x-button")).toHaveClass("bg-green-900");
    expect(screen.getByTestId("2.0x-button")).not.toHaveClass("bg-green-900");
  });

  test("speed set to 2.0x", () => {
    const speed: number = 2;
    render(<SpeedControl speed={speed} setSpeed={setSpeed} />);

    expect(screen.getByTestId("1.0x-button")).not.toHaveClass("bg-green-900");
    expect(screen.getByTestId("1.5x-button")).not.toHaveClass("bg-green-900");
    expect(screen.getByTestId("2.0x-button")).toHaveClass("bg-green-900");
  });

  test("snapshot test", () => {
    const speed: number = 1;
    const { asFragment } = render(
      <SpeedControl speed={speed} setSpeed={setSpeed} />
    );
    expect(asFragment).toMatchSnapshot();
  });
});
