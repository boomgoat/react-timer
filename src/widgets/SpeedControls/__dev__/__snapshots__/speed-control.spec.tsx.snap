// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speed Control Tests snapshot test 1`] = `[Function]`;
